### DISCUSSION TITLE
Cross filter is not showing up in Superset UI, after enabled in config.py

### DISCUSSION THREAD

**Discussion by nagarajmmu** [apache/superset#22345] [Q&A / Help]
Hi Team

I have installed a superset latest version using Docker in Linux (Centos) machine.

While going through the below video, I came to know that CROSS filter option is available in superset., please find the link for reference.
[Introducing Advanced Drill-Down Functionality In Apache Superset Using Apache Echarts](https://www.youtube.com/watch?v=7YnpKLZ1PRM)

I have observed "DASHBOARD_CROSS_FILTERS" is available in superset config.py, I have enabled it, but CROSS FILTER is not showing in chart UI.

Please let me know, how can I get CROSS FILTER in superset. I have been waiting for CROSS FILTER and DRILL DOWN feature from last one year.

#### How to reproduce the bug

1. Got to Charts Menu, 
2. Click on New Chart
3. Choose a Dataset
4. Select ECharts
5. Select Pie Chart.
6. Click on Create New Chart

### Expected results

In Pie Chart, under "DATA" tab, "ENABLE DASHBOARD CROSS FILTETS" checkbox option should be available below the "FILTERS", but I am not able to see "ENABLE DASHBOARD CROSS FILTETS" checkbox.

### Actual results

In Pie Chart, under "DATA" tab, "ENABLE DASHBOARD CROSS FILTETS" checkbox option is not available.

#### Screenshots

![image](https://user-images.githubusercontent.com/61146310/205734519-66db97f5-99b3-4658-9d74-46c8e34a64e0.png)

### Environment

- browser type and version: Chrome - v108.0.5359.71
- superset version: `0.0.0-dev (Latest)`
- python version: `3.8`
- any feature flags active: Yes below flags are enabled
              "DASHBOARD_NATIVE_FILTERS": True,
               "DASHBOARD_CROSS_FILTERS": True

Please let me know, how can I get CROSS FILTER, DRILL DOWN features in superset, I am fine to add plugins if available. 

Thanks
Nagaraj M M

**Comment by kamalkeshavani-aiinside** [comment]
Have you tried running `superset init` command after changing the config.py? Usually some changes in the config, require execution of this command to bring the changes to UI.

**Comment by rusackas** [comment]
Hi @nagarajmmu. I'm hopeful that @kamalkeshavani-aiinside's help might steer you in the right direction. The `DASHBOARD_CROSS_FILTERS` feature flag is indeed exactly the one you're looking for. If you're using `docker-compose` then it should just automatically update and show in the control panel as you expect. Since this does seem to be more of a configuration troubleshooting issue than a bug report, per se, I'm going to go ahead and move this to a Q&A Discussion.

**Comment by nagarajmmu** [comment]
Hi Kamal and Evan

Thanks for the response, as I am using Docker, Docker is having superset-init as a part of start, as shown below.
So superset init will execute, when I start superset,
superset-init:
  image: *superset-image
  container_name: superset_init
  command: ["/app/docker/docker-init.sh"]
  env_file: docker/.env-non-dev
  depends_on: *superset-depends-on
  user: "root"
  volumes: *superset-volumes
  healthcheck:
    disable: true

But still facing the same issue, is there any other way to confirm, or any log that says, CROSS FILTER is not enabled or having issue.

For your information, I am using, below command to start the Superset Docker
--docker-compose -f docker-compose-non-dev.yml up

Please let me know, how to verify that CROSS FILTER is enabled.

Thanks
Nagaraj M M

**Reply by rusackas** [reply]
One way is to inspect the HTML of the page in the browser. You'll see a giant block of bootstrap data near the top that should include the settings of ALL the feature flags (the frontend reads and interprets these). If you see it in there as True, then... it's true. That said, I'd be quite shocked if it's True in the bootstrap payload, but not being reflected in the UI.

**Reply by nagarajmmu** [reply]
Hi Evan

Thanks for the information.
I verified the HTML data bootstrap, I found that DASHBOARD_CROSS_FILTERS is false. Attached screenshot.
![image](https://user-images.githubusercontent.com/61146310/206005118-af1a261c-7e77-416c-92ff-fba45d5e4c15.png)

I have changed DASHBOARD_CROSS_FILTERS to True in /superset/superset/config.py and rebuild the docker and start the docker with below command.
--docker-compose -f docker-compose-non-dev.yml up

Looks like something is missing to enable DASHBOARD_CROSS_FILTERS in superset.

Please let me know, if I did anything wrong here.

Thanks
Nagaraj M M

**Reply by nagarajmmu** [reply]
Hi Evan

What would have went working in enabling DASHBOARD_CROSS_FILTERS in superset.
Do you have any thoughts on this.

Thanks
Nagaraj M M

**Reply by rusackas** [reply]
Honestly, I'm fairly stumped here as to why the flag is clearly not enabled in bootstrap data, if you've indeed flipped the flag in config.py. Maybe @villebro or @nytai have some idea of how to kick the darn thing when it's stuck?

**Reply by Shisir99** [reply]
Facing the same issue , Have any of you guys resolved it? @nagarajmmu @rusackas @frafra @ryw

**Reply by nagarajmmu** [reply]
Hi Shisir99

I face this issue, when I installed superset from master branch.
So I removed it and installed it from the stable release from here https://github.com/apache/superset/releases

now issue resolved and working fine.

Thanks
Nagaraj M M

**Reply by therahilshaikh** [reply]
Hi @nagarajmmu I followed the same steps 
Installed stable release from provided link and updated value of DASHBOARD_CROSS_FILTERS to True, but still it's not working.
Can anyone help me resolving this?

**Reply by nagarajmmu** [reply]
Hi therahilshaikh

I have installed in Ubuntu 18 machine, cross filter working fine for me, also cross filter will work only for ECharts.

Thanks
Nagaraj M M

**Reply by therahilshaikh** [reply]
Hi @nagarajmmu ,
When i am checking by inspecting page in the header section of page, DASHBOARD_CROSS_FILTERS shown as False.

**Reply by Unanimad** [reply]
Hello everyone! I'm trying to enable `DASHBOARD_CROSS_FILTERS` too.  
Using master branch and superset-helm-chart-0.8.8 tag not working as well. Also 2.1.0rc3.

![image](https://user-images.githubusercontent.com/5610653/227188460-f6baa0b5-a75e-415f-a11c-fd897ba7f66f.png)
![image](https://user-images.githubusercontent.com/5610653/227188576-cbdd3be0-f8d0-4060-855d-c940208339cd.png)

**Reply by JohannesSchnell** [reply]
i made it work by
- adding `"DASHBOARD_CROSS_FILTERS": True` to `FEATURE_FLAGS` in[ superset_config.py](https://github.com/apache/superset/blob/master/docker/pythonpath_dev/superset_config.py#L103)  that makes the change visable in the html page @therahilshaikh 


activating the feature on an active dashboard, this took a while to figure out: 
- open a dashboard
- on the left site open the filter frame (click the arrow -->|) 
- click on the 'options' wheel 
- check 'enable cross-filtering'

@Unanimad 

![Bildschirmfoto vom 2023-03-27 14-12-39](https://user-images.githubusercontent.com/108523161/227939406-542a9755-8c73-47fd-9f24-20fc6c17b0c8.png)

**Reply by Unanimad** [reply]
Thank you! I going on try it. As soon as possible I will report here

**Reply by mdeshmu** [reply]
Its most likely you have not enabled feature flag. Please share your specific config section to see how you have done it.
In Superset versions before 2.1.0, dashboard cross filters are to be enabled at chart level and are available only for select charts.
In 2.1.0, this feature has been moved to dashboard and you no longer have to enable it in every chart. I hope that helps.

**Reply by Unanimad** [reply]
> enabled feature flag

Sorry for my delay,

```
LANGUAGES = {
    'pt_BR': {'flag': 'br', 'name': 'Brazil'},
    'en': {'flag': 'us', 'name': 'English'},
    'fr': {'flag': 'fr', 'name': 'French'},
}

FEATURE_FLAGS = {'DASHBOARD_CROSS_FILTERS': True}
DASHBOARD_RBAC = True
BABEL_DEFAULT_LOCALE = 'pt_BR'
BABEL_DEFAULT_TIMEZONE = 'America/Recife'

PUBLIC_ROLE = 'Public'
```

![image](https://user-images.githubusercontent.com/5610653/231202173-c4adc6e6-bf98-46da-9a1c-bfe9c907adb7.png)

**Reply by suregoodru** [reply]
Hi! I have similar problem in v. 2.1.0
<img width="343" alt="image" src="https://user-images.githubusercontent.com/4560198/232772673-3a83db76-cd4b-4756-80ac-97aec4ea698d.png">

I see option in dashboard settings, but not see in chart settings
<img width="264" alt="image" src="https://user-images.githubusercontent.com/4560198/232772806-ff13b63f-d81e-45b0-8d8e-2938735b9c6e.png">

**Reply by TehsimAftab** [reply]
Hello,

With this option on chart click the filter will be on the page, with with (enable dashboard cross filter) from pie chart we can dill down to anther details of the pie chart which s very useful. My Superset Version is 2.1.0 and sqlparse is 0.4.3 but still i cannot see that checkbox its missing.

**Reply by harrypotter033** [reply]
how do i we enable in helm chart values.yaml, i am using 0.10.6 chart and 2.1.0 superset

**Comment by harrypotter033** [comment]
Hi everyone! I'm facing an issue with Apache Superset deployed using Helm. I've tried enabling the "dashboard cross filter" feature by setting the feature flags DASHBOARD_NATIVE_FILTERS and DASHBOARD_CROSS_FILTERS to true. I've updated my Helm deployment accordingly, but I'm not seeing the cross-filtering option in the UI.
Details:
Superset Version: 2.1.0
Helm Chart Version: 0.10.6
Configuration Changes:
 Values.yaml
extraEnv: FEATURE_FLAGS: '{"DASHBOARD_CROSS_FILTERS": true, "DASHBOARD_NATIVE_FILTERS": true}'
I've already verified that I'm on a compatible Superset version, checked browser caching, and ensured that the config changes propagated to the application. Does anyone have any suggestions or has encountered this issue before? Thanks in advance! ![image](https://github.com/apache/superset/assets/4506144/3a350ee0-1bd0-47a9-9e54-57617009b0f3)

**Comment by AvinashGupta** [comment]
I am having the same issue with 3.1.0. Not able to see cross filtering option for charts. It's only visible for dashboards. 
<img width="429" alt="image" src="https://github.com/apache/superset/assets/5013292/be66b926-c970-40cf-8b76-65e4bc2336ed">

**Reply by rusackas** [reply]
Not sure what you mean... looks like it's enabled in your screenshot. Try clicking a wedge on a Pie Chart or something. It's a dashboard-level feature, so you should only see it in dashboards.
